examples/client/object_server.c

```diff
@@ -224,6 +224,50 @@ static uint8_t prv_get_value(lwm2m_data_t * dataP,
     }
 }
 
+static void remove_optional_resource(uint16_t *resList, int val, int id, int *nbRes) {
```

Comment:
I know this is a matter of taste, but I think using early exits and therefore keeping the happy path as little nested as possible, helps readability:

```
static void remove_optional_resource(uint16_t *const resList, int val, int id, int *nbRes) {
    if (val >= 0) {
        /* some explanation on why quitting here */
        return;
    }

    for (int i = 0; i < *nbRes; i++) {
        if (resList[i] == id) {
            *nbRes -= 1;
            memmove(&resList[i], &resList[i + 1], (*nbRes - i) * sizeof(resList[i]));
            break;
        }
    }
}
```